lib/active_merchant/billing/gateways/paypal_checkout.rb

```diff
@@ -0,0 +1,376 @@
+require 'active_merchant/billing/gateways/paypal/paypal_checkout_common'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class PaypalCheckoutGateway < Gateway
+      include PaypalCheckoutCommon
+      
+      self.supported_countries = ['AU', 'AT', 'BE', 'BG', 'CA', 'CY', 'CZ', 'DK', 'EE', 'FI', 'FR', 'GR', 'HU', 'IT', 'LV', 'LI', 'LT', 'LU', 'MT', 'NL', 'NO', 'PL', 'PT', 'RO', 'SK', 'SI', 'ES', 'SE', 'US', 'GB']
+      self.homepage_url        = 'https://www.paypal.com/us/business/platforms-and-marketplaces'
+      self.display_name        = 'PayPal Commerce Platform'
+      self.default_currency    = 'USD'
+      self.supported_cardtypes = [:visa, :master, :american_express, :discover, :jcb]
+
+      def create_order(intent, options)
+        requires!(options.merge!(intent.nil? ? {} : { intent: intent}), :intent, :purchase_units)
+
+        post = {}
+        add_intent(intent, post)
+        add_purchase_units(options[:purchase_units], post)
+        add_payment_instruction(options[:payment_instruction], post) unless options[:payment_instruction].blank?
+        add_application_context(options[:application_context], post) unless options[:application_context].blank?

```

Comment:
what should be passed in application context? is this the browser details and stuff for fraud?

Reply:
let's mark things that are required fields and the ones we'll use. otherwise, it going to be difficult for us to review the PR in integrations.